### ISSUE TITLE
`mean_horizontal` with `List` or `Array` columns causes `PanicException`

### ISSUE THREAD

**Issue by nik-sm** [pola-rs/polars#19467]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

Using `List` schema:
```python
import polars as pl
df = pl.DataFrame(
    {"experiment_id": [1, 2], "sensor1": [[1,2,3], [7,8,9]], "sensor2": [[4,5,6],[10,11,12]]}
)
df.with_columns(pl.mean_horizontal("sensor1", "sensor2").alias("avg_sensor"))
```

Using `Array` schema:
```python
import polars as pl
df = pl.DataFrame(
    {"experiment_id": [1, 2], "sensor1": [[1,2,3], [7,8,9]], "sensor2": [[4,5,6],[10,11,12]]},
    schema={"experiment_id": pl.Int64, "sensor1": pl.Array(pl.Int64, 3), "sensor2": pl.Array(pl.Int64, 3)}
)
df.with_columns(pl.mean_horizontal("sensor1", "sensor2").alias("avg_sensor"))
```

### Log output

```shell
thread 'polars-2' panicked at crates/polars-core/src/frame/mod.rs:2859:26:
called `Option::unwrap()` on a `None` value
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "PATH_TO_VENV/lib/python3.12/site-packages/polars/dataframe/frame.py", line 9194, in with_columns
    return self.lazy().with_columns(*exprs, **named_exprs).collect(_eager=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "PATH_TO_VENV/lib/python3.12/site-packages/polars/lazyframe/frame.py", line 2055, in collect
    return wrap_df(ldf.collect(callback))
                   ^^^^^^^^^^^^^^^^^^^^^
pyo3_runtime.PanicException: called `Option::unwrap()` on a `None` value
```

### Issue description

I have a dataframe where each row represents an experiment.
I have several `List` columns to hold the timeseries outputs from several sensor channels.
I want to use `DataFrame.mean_horizontal()` to compute the average timeseries and store this in a new column.

I can see that `mean_horizontal` accepts columns with `List` dtype by trying the following (trivial) query that "averages" a single `List` column. This query does not produce an error:
```python
import polars as pl
df = pl.DataFrame({"a": [[1,2,3], [4,5,6]]})
df.with_columns(pl.mean_horizontal("a").alias("dummy_average"))
```

If this is not the intended approach, I would really appreciate guidance on the intended way to get an average timeseries. Also - if this is not intended/supported usage, I think that `mean_horizontal` should throw an error when receiving a column with `List` or `Array` dtype.

### Expected behavior

Averaging two `List` columns (or two `Array` columns of matching shape) should produce a single `List` column, which is the elementwise average of the two. 

Edge-case behavior needs to be defined for when `List` entries of the two columns do not have the same length. Either this is an error, or the missing value from the shorter list is ignored when computing the average.

### Installed versions

<details>

```
>>> pl.show_versions()
--------Version info---------
Polars:              1.11.0
Index type:          UInt32
Platform:            macOS-15.0.1-arm64-arm-64bit
Python:              3.12.6 (main, Sep 27 2024, 17:59:45) [Clang 15.0.0 (clang-1500.3.9.4)]
LTS CPU:             False

----Optional dependencies----
adbc_driver_manager  <not installed>
altair               <not installed>
cloudpickle          <not installed>
connectorx           <not installed>
deltalake            <not installed>
fastexcel            <not installed>
fsspec               <not installed>
gevent               <not installed>
great_tables         <not installed>
matplotlib           <not installed>
nest_asyncio         <not installed>
numpy                <not installed>
openpyxl             <not installed>
pandas               <not installed>
pyarrow              <not installed>
pydantic             <not installed>
pyiceberg            <not installed>
sqlalchemy           <not installed>
torch                <not installed>
xlsx2csv             <not installed>
xlsxwriter           <not installed>
```

</details>

**Comment by nik-sm** [comment]
For the time-being, here's a workaround I found:


```python
import polars as pl

df = pl.DataFrame({"experiment_id": [1, 2], "sensor1": [[1, 2, 3], [7, 8, 9]], "sensor2": [[4, 5, 6], [10, 11, 12]]})

avg_sensor = (
    df.explode(["sensor1", "sensor2"])
    .with_columns(pl.mean_horizontal("sensor1", "sensor2").alias("avg_sensor"))
    .group_by("experiment_id")
    .agg(pl.col("avg_sensor"))
)

result = df.join(avg_sensor, on="experiment_id")
print(result)
```

**Comment by cmdlineluser** [comment]
`sum_horizontal` doesn't seem to error.

```python
df.with_columns(
    pl.sum_horizontal("sensor1", "sensor2").alias("avg_sensor") / 2
)

# shape: (2, 4)
# ┌───────────────┬───────────┬──────────────┬──────────────────┐
# │ experiment_id ┆ sensor1   ┆ sensor2      ┆ avg_sensor       │
# │ ---           ┆ ---       ┆ ---          ┆ ---              │
# │ i64           ┆ list[i64] ┆ list[i64]    ┆ list[f64]        │
# ╞═══════════════╪═══════════╪══════════════╪══════════════════╡
# │ 1             ┆ [1, 2, 3] ┆ [4, 5, 6]    ┆ [2.5, 3.5, 4.5]  │
# │ 2             ┆ [7, 8, 9] ┆ [10, 11, 12] ┆ [8.5, 9.5, 10.5] │
# └───────────────┴───────────┴──────────────┴──────────────────┘
```

**Comment by nik-sm** [comment]
Good find. It's a cleaner workaround as well - thank you

**Comment by VillePuuska** [comment]
@ritchie46 Wrt the fix in PR #19472, would it make sense to allow `mean_horizontal` to also work on lists of numerical values since the "manual" workaround with `sum_horizontal` works on them? If yes, I think it's as simple as changing the `dtype` check [here](https://github.com/pola-rs/polars/blob/687811d43ccfd95e3db846545b3bb41136993d2b/crates/polars-core/src/frame/mod.rs#L2851) to filter only numerical columns to

```rust
dtype.is_numeric()
    || matches!(dtype, DataType::Boolean)
    || (dtype.is_list()
        && (dtype.inner_dtype().unwrap().is_numeric()
            || matches!(dtype.inner_dtype().unwrap(), DataType::Boolean)))
```

For arrays this wouldn't work since dividing an array by a float wont work as is, but for lists this seems to work.

**Comment by VillePuuska** [comment]
Could be as simple as [this](https://github.com/VillePuuska/polars/commit/ab5657f020d964a833641d91ab38ce3c1bbfa312).
